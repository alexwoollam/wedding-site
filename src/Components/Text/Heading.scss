@import '../../variables';

.heading{
  &-text{
    font-size: 2.5rem;
    font-weight: 700;
    line-height: 1.5;
    padding: 0;
    color: $dustypink;
  }
  &-subtext{
    display: inline-block;
    width: 100%;
    text-align: center;
    font-size: 0.75rem;
    font-weight: 500;
    line-height: 1.5;
    padding: 0;
    color: $dustypink;
  }
  margin: 60px 0;
}