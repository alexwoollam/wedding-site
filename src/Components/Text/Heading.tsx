import React from 'react';

interface TextSent {
    text: string;
}

function Heading( props:TextSent ) {
  if( ! props.text ){
    return null;
  }
  return (
    <h1 
      className="text-grey-700 font-serif text-3xl text-center m-5"
      data-testid="text-block"
    >
        { props.text }
    </h1>
  );
}

export default Heading;