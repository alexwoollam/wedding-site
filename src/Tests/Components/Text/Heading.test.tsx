import '@testing-library/jest-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Heading from '../../../Components/Text/Heading';


describe('<Heading/> component', () => {
  test('renders keep me updated form', () => {
    render(<Heading>Heading</Heading>);
    const linkElement = screen.getByText('Heading');
    expect(linkElement).toBeInTheDocument();
  });

  test('If no props are sent, do an early return.', () => {
    render(<Heading></Heading>);
    const linkElement = screen.queryByTestId('heading-block');
    expect(linkElement).not.toBeInTheDocument();
  });
  
  it('renders correctly', () => {
    const tree = renderer
        .create(<Heading>Heading text</Heading>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
