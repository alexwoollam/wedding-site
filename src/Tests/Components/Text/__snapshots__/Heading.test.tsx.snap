// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> component renders correctly 1`] = `
<h1
  className="text-grey-700 font-serif text-3xl text-center m-2"
  data-testid="text-block"
>
  Heading text
</h1>
`;
